sensors/imu/lsm9dsxx.c

```diff
@@ -2,87 +2,204 @@
+#define VAL_CTRL_REG6_XL_FS_8G      0x18 /* strange order, not a mistake! */
+#define VAL_CTRL_REG6_XL_FS_16G     0x08 /* strange order, not a mistake! */
+
+/* control register 8 */
+#define REG_CTRL_REG8 0x22
+
+/* control register 9 */
+#define REG_CTRL_REG9         0x23
+#define VAL_CTRL_REK9_I2C_DIS 0x04
+
+/* data storage addresses */
+#define REG_DATA_OUT_TEMP 0x15
+#define REG_DATA_OUT_GYRO 0x18
+#define REG_DATA_OUT_ACCL 0x28
+#define SPI_READ_BIT      0x80
+
+/* sensors data sizes */
+#define SENSOR_OUTPUT_SIZE 6
+#define GYR_OVERFLOW       26820 /* if gyro returns more than this, the sensorhub result will overflow */
+
+/* conversions */
+#define GYR2000DPS_CONV_MRAD 1.221730475F /* convert gyroscope value (at scale 2000DPS) to mrad/s */
+#define ACC8G_CONV_MS2       0.244F       /* convert accelerations (at 8G scale) [m/s^2] */
+
+
 #include <errno.h>
 #include <stdlib.h>
 #include <unistd.h>
 #include <sys/time.h>
 #include <sys/threads.h>
+#include <spi.h>
+#include <spi-msg.h>
+#include <string.h>
 
 #include "../sensors.h"
 
 
 typedef struct {
+	spimsg_ctx_t spiCtx;
 	sensor_event_t evtAccel;
 	sensor_event_t evtGyro;
 	char stack[512] __attribute__((aligned(8)));
 } lsm9dsxx_ctx_t;
 
 
-static inline time_t lps25xx_timeMsGet(void)
+static int16_t translateGyr(uint8_t hbyte, uint8_t lbyte)
 {
-	time_t now;
-	gettime(&now, NULL);
+	int16_t val = 0;
+	val |= hbyte;
+	val <<= 8;
+	val |= lbyte;
 
-	return now / 1000;
+	/* overflow handling */
+	if (val > GYR_OVERFLOW) {
+		return INT16_MAX;
+	}
+	if (val < -GYR_OVERFLOW) {
+		return INT16_MIN;
+	}
+	return GYR2000DPS_CONV_MRAD * val; /* sensor value to [mrad/s] */
+}
+
+
+static int16_t translateAcc(uint8_t hbyte, uint8_t lbyte)
+{
+	int16_t val = 0;
+	val |= hbyte;
+	val <<= 8;
+	val |= lbyte;
+	return ACC8G_CONV_MS2 * val; /* sensor value to [mm/s^2] */
+}
+
+
+int spiWriteReg(spimsg_ctx_t *spiCtx, uint8_t regAddr, uint8_t regVal)
+{
+	unsigned char cmd[2] = { (regAddr & 0x7F), regVal }; /* write bit set to regAddr */
+
+	return spimsg_xfer(spiCtx, cmd, sizeof(cmd), NULL, 0, sizeof(cmd));
+}
+
+
+static int lsm9dsxx_hwSetup(spimsg_ctx_t *spiCtx)
+{
+	/* auto increment + SW reset */
+	if (spiWriteReg(spiCtx, REG_CTRL_REG8, 0x05) < 0) {
+		return -1;
+	}
+	usleep(1000 * 100);
+
+	/* ranges and sampling of accelerometer and gyro */
+	if (spiWriteReg(&spiCtx, REG_CTRL_REG1_G, (VAL_CTRL_REG1_G_ODR_G_952HZ | VAL_CTRL_REG1_G_FS_G_2000)) < 0) {
+		return -1;
+	}
+	if (spiWriteReg(spiCtx, REG_CTRL_REG6_XL, (VAL_CTRL_REG6_XL_ODR_XL_952 | VAL_CTRL_REG6_XL_FS_8G)) < 0) {
+		return -1;
+	}
+	usleep(1000 * 100);
+
+	/* enabling sensors */
+	if (spiWriteReg(spiCtx, REG_CTRL_REG5_XL, (VAL_CTRL_REG5_XL_ENABLE)) < 0) {
+		return -1;
+	}
+	if (spiWriteReg(spiCtx, CTRL_REG4, (CTRL_REG4_G_ENABLE)) < 0) {
+		return -1;
+	}
+	usleep(1000 * 100);
+
+	return 0;
 }
 
 
 static void lsm9dsxx_threadPublish(void *data)
 {
+	int err;
+	uint64_t tstamp_gyro, tstamp_accl;
```

Comment:
```suggestion
	time_t tstamp_gyro, tstamp_accl;
```
`gettime()` expects `time_t` type, also in GPS @damianloew  has changed `timestamp` time from `uint64_t` to `time_t`, check it out:
https://github.com/phoenix-rtos/phoenix-rtos-devices/blob/3f373779f4d86620b856fa2877bea65927cde88f/sensors/include/libsensors.h#L102